faster_whisper/transcribe.py

```diff
@@ -80,6 +86,244 @@ class TranscriptionInfo(NamedTuple):
+
+
+    def transcribe(
+        self, audio: Union[str, np.ndarray], vad_segments, batch_size=None, num_workers=0, language=None, task=None, print_progress = False, combined_progress=False
+    ) -> TranscriptionResult:
+        if isinstance(audio, str):
+            audio = decode_audio(audio)
+
+        def data(audio, segments):
+            for seg in segments:
+                f1 = int(seg['start'] * self.model.feature_extractor.sampling_rate)
+                f2 = int(seg['end'] * self.model.feature_extractor.sampling_rate)
+                yield {'inputs': audio[f1:f2]}
+
+        language, task = self.get_language_and_tokenizer(audio, task, language)
+                
+        if self.suppress_numerals:
+            previous_suppress_tokens = self.options.suppress_tokens
+            numeral_symbol_tokens = find_numeral_symbol_tokens(self.tokenizer)
+            print(f"Suppressing numeral and symbol tokens: {numeral_symbol_tokens}")
+            new_suppressed_tokens = numeral_symbol_tokens + self.options.suppress_tokens
+            new_suppressed_tokens = list(set(new_suppressed_tokens))
+            self.options = self.options._replace(suppress_tokens=new_suppressed_tokens)
+
+        segments: List[SingleSegment] = [] 
+
+        batch_size = batch_size or self._batch_size
+        total_segments = len(vad_segments)
+        for idx, out in enumerate(self.__call__(data(audio, vad_segments), batch_size=batch_size, num_workers=num_workers)):
+            if print_progress:
+                base_progress = ((idx + 1) / total_segments) * 100
+                percent_complete = base_progress / 2 if combined_progress else base_progress
+                print(f"Progress: {percent_complete:.2f}%...")
+
+            text = out['text']
+            if batch_size in [0, 1, None]:
+                text = text[0]
+
+            segments.append({
+                'text':text,
+                'start': round(vad_segments[idx]['start'], 3),
+                'end': round(vad_segments[idx]['end'], 3)
+            })
+
+        # revert the tokenizer if multilingual inference is enabled
+        if self.preset_language is None:
+            self.tokenizer = None
+
+        # revert suppressed tokens if suppress_numerals is enabled
+        if self.suppress_numerals:
+            self.options = self.options._replace(suppress_tokens=previous_suppress_tokens)
+
+        return {"segments": segments, "language": language} 
+    
+    def detect_language(self, audio: np.ndarray):
+        
+        segment = torch.tensor(self.model.feature_extractor(audio, padding=True)[:,:self.model.feature_extractor.nb_max_frames])
+        encoder_output = self.model.encode(segment)
+        results = self.model.model.detect_language(encoder_output)
+        language_token, language_probability = results[0][0]
+        language = language_token[2:-2]
+        print(f"Detected language: {language} ({language_probability:.2f}) in first 30s of audio...")
```

Comment:
Removed all print statements and rerouted to logger (info&warning/debug) accordingly. 
The function now also returns language_probability, although that's not very important since we have a more reliable method for language identification (`detect_language_multi_segment`).